### PR TITLE
add storage upload file

### PR DESCRIPTION
**Pull request by bshaffer**
addresses #8

### CODE REVIEW

File: storage/lib/storage_samples.ex [hunk 1]
```diff
@@ -20,4 +20,35 @@ defmodule GoogleApi.Storage.Samples do
...
+  List storage buckets for a project.
+
+  ## Examples
+
+      iex> GoogleApi.Storage.Samples.upload_file("bucket_id", "test/file.txt")
+      Uploaded file.text to  https://www.googleapis.com/storage/v1/b/bucket_id/o/file.txt
+      your_project_bucket2
+
+  """
+  def upload_file(bucket_id, file) do
```
Comment (chingor13) [inline storage/lib/storage_samples.ex]: Maybe to clarify, rename to `file_fullpath` so the reader doesn't think it's some kind of file struct.

File: storage/test/storage_samples_test.exs [hunk 2]
```diff
@@ -1,7 +1,11 @@
 defmodule GoogleApi.Storage.Samples.Test do
   use ExUnit.Case
   import ExUnit.CaptureIO
-  doctest GoogleApi.Storage.Samples, except: [:moduledoc, list_buckets: 1]
+  doctest GoogleApi.Storage.Samples, except: [
```
Comment (chingor13) [inline storage/test/storage_samples_test.exs]: Let's just remove this if we're ignoring all the doctests.

File: storage/lib/storage_samples.ex (hunk 1 above)
Comment (bshaffer) [reply storage/lib/storage_samples.ex]: Relative file paths also work, so how about `file_path`?

File: storage/test/storage_samples_test.exs [hunk 3]
```diff
@@ -11,4 +10,14 @@ defmodule GoogleApi.Storage.Samples.Test do
     end)
     assert String.contains? output, project_id
   end
+
+  test "upload object" do
+    # Use the project ID as the bucket ID
+    bucket_id = System.get_env("GOOGLE_PROJECT_ID")
+    assert bucket_id
+    output = capture_io(fn ->
+        GoogleApi.Storage.Samples.upload_file(bucket_id, "test/data/file.txt")
```
Comment (dazuma) [inline storage/test/storage_samples_test.exs]: indent is 2 spaces

File: storage/lib/storage_samples.ex [hunk 4]
```diff
@@ -20,4 +20,35 @@ defmodule GoogleApi.Storage.Samples do
...
+      conn,
+      bucket_id,
+      "multipart",
+      %{name: Path.basename(file_path)},
+      file_path
+    )
+
+    # Print the object.
+    IO.puts("Uploaded #{object.name} to #{object.selfLink}")
+    "done!"
```
Comment (dazuma) [inline storage/lib/storage_samples.ex]: What is this return value for?

**Pull request merged**